### ISSUE TITLE
MySQL connector does not automatically refresh source schema when changing source

### ISSUE THREAD

**Issue by iekdosha** [airbytehq/airbyte#40235]
### Connector Name

source-mysql

### Connector Version

MySQL: 3.4.7
Snowflake: 3.9.1

### What step the error happened?

Updating the connector

### Relevant information

Hi
I am testing schema change propagation from MySQL to Snowflake (with CDC) and when I add a column it does not automatically replicate that change even tough I have the "Propagate all streams and fields" setting defined:
![image](https://github.com/airbytehq/airbyte/assets/35876294/a86a8af0-ecc4-4246-9dd6-7bab54c51e81)

- There have been several syncs since I have changed this config.
- sync mode is "Incremental | Append + Deduped"
- I have tried restarting my local Airbyte deployment
- The new column will be detected if I manually press the "refresh the source schemas" button, at which point any data changed since I introduced the field is not replicated

Thanks :)

### Contribute

- [ ] Yes, I want to contribute

**Comment by marcosmarxm** [comment]
The version reported is not the version of the connector. @iekdosha please update it.

**Comment by iekdosha** [comment]
@marcosmarxm Thank you for informing me. I have updated to the correct version and also included the target connector version, just in case there's an issue related to the target.

**Comment by evantahler** [comment]
At the moment, schema refreshes happen once-per-day, so you'll need to wait 24h to be notified.  We'll be increasing that frequency later this year.

**Comment by stim371** [comment]
> At the moment, schema refreshes happen once-per-day, so you'll need to wait 24h to be notified. We'll be increasing that frequency later this year.

@evantahler is there any update on this?

Chatting with the AI in slack it sounds like the workaround is to follow these steps with the deprecated API: https://github.com/airbytehq/airbyte/issues/26399 But for sources with multiple connections it can get hairy.

Would be wonderful to have a built-in refresh API.

**Comment by evantahler** [comment]
cc @theyueli

**Comment by stim371** [comment]
@evantahler @theyueli After switching my connection's "Detect and propagate schema changes" from "Manually approve" to "Propagate fields only", it appears that `GET /streams` and `POST /discover_schemas` both kick off a full schema refresh from the source, regardless of whether you pass `ignoreCache=true`.

It seems like it does what I was hoping already (although it's a little surprising behavior, and also appears to bypass the cache all the time).

**Comment by theyueli** [comment]
Yes, the setting used by @stim371 allows detection of schema change. In the worst case, after detecting a schema change, we will always ask user if they want to do full refresh. 

@iekdosha, by the way, we have rolled out a new generation of source MySQL. Please feel free to try the latest version and pay attention to the settings mentioned above. 

Close this issue.
